.gitlab-ci.yml

```diff
@@ -30,50 +126,45 @@ Test:
   - dev
   - master
   - /^feature\/.*$/
+  - /^migration\/.*$/
 
+# Job to build static asset for staging and live environment
+# dev branch -> Staging
+# master branch -> Live
 Test_N_Build:
-  image: node:12.13.0
-  stage: test_build_static
+  extends: .build-static
 
-  before_script:
-  - cd src/ensembl
-  - npm ci
+  variables:
+    GOOGLE_ANALYTICS_KEY: ${GA_KEY}
+    ENVIRONMENT: production
+    API_HOST: ""
 
   script:
   - npm run test
   - export NODE_ENV=production
-  - export GOOGLE_ANALYTICS_KEY=${GA_KEY}
-  - export ENVIRONMENT=production
-  - export API_HOST=""
   - npm run build
 
-  artifacts:
-    name: static_assets
-    paths:
-    - src/ensembl/dist/
-
   only:
   - dev
   - master
 
+# Job to build static asset for internal environment
+# dev branch -> Internal
 Test_N_Build:internal:
-  image: node:12.13.0
-  stage: test_build_static
+  extends: .build-static
 
-  before_script:
-  - cd src/ensembl
-  - npm ci
+  variables:
+    GOOGLE_ANALYTICS_KEY: ${GA_KEY}
+    ENVIRONMENT: internal
+    API_HOST: ""
 
   script:
   - npm run test
   - export NODE_ENV=development
-  - export GOOGLE_ANALYTICS_KEY=${GA_KEY}
-  - export ENVIRONMENT=internal
-  - export API_HOST=""
   - npm run build
 
   artifacts:
-    name: static_assets_internal
+    name: static_assets
     paths:
     - src/ensembl/dist/
 
```

Comment:
Below, in `Test_N_Build:feature`, `NODE_ENV=development` should also be replaced with `NODE_ENV=production`.

Reply:
I would be doing another major refactoring for Feature related jobs later as this will need redeployment of all the feature branch with different tags. This may hamper the development so I will be doing it later.